src/k2/cmd/tpcc/transactions.h

```diff
@@ -50,8 +87,18 @@ class AtomicVerify;
 
 class TPCCTxn {
 public:
-    virtual future<bool> run() = 0;
+    virtual future<bool> attempt() = 0;
     virtual ~TPCCTxn() = default;
+
+    future<bool> run() {
+        // retry 10 times for a failed transaction
+        auto retryStrategy = make_shared<FixedRetryStrategy>(10);
```

Comment:
You can make this a unique_ptr and capture it by move in a "finally" block to safely dispose of it.

Reply:
I was going to suggest to use a do_with block here instead of the pointer create and capture trick

Reply:
changed to use do_with block as suggested.